### PR TITLE
[PLGN-222] Servicenow - Updated the vulnerability source from Application Vulnerability Item to Vulnerability Item

### PR DESCRIPTION
**Pull request by igorski-r7**
### Description

Describe the proposed changes:

  - Updated the vulnerability source from Application Vulnerability Item to Vulnerability Item

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [X] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [X] Screenshot of job output with the plugin changes
- [X] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [X] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [X] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [X] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [X] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [X] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [X] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [X] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [X] Work fully completed
- [X] Functional
  - [X] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [X] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [X] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [X] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [X] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

**Review by hpwilson-r7 — APPROVED** [review]

File: plugins/servicenow/plugin.spec.yaml [hunk 1]
```diff
@@ -670,12 +681,42 @@ actions:
...
         example: Open
+      source:
+        title: Source
+        description: The vulnerability source
+        type: string
+        required: false
+        example: ExampleSource
+      vulnerability:
+        title: Vulnerability
+        description: The reference of vulnerability found
```
Comment (ablakley-r7) [inline plugins/servicenow/plugin.spec.yaml:692]:
```suggestion
        description: The reference of the found vulnerability
```

File: plugins/servicenow/plugin.spec.yaml [hunk 2]
```diff
@@ -670,12 +681,42 @@ actions:
...
+        type: string
+        required: false
+        example: 9de5069c5afe602b2ea0a04b66beb2c0
+      ip_address:
+        title: IP Address
+        description: The IP address of source the vulnerability was found
```
Comment (ablakley-r7) [inline plugins/servicenow/plugin.spec.yaml:698]:
```suggestion
        description: The IP address of the source where the vulnerability was found
```

File: plugins/servicenow/plugin.spec.yaml [hunk 3]
```diff
@@ -670,12 +681,42 @@ actions:
...
+        type: string
+        required: false
+        example: 192.168.0.1
+      dns:
+        title: DNS Name
+        description: The name of source DNS the vulnerability was found
```
Comment (ablakley-r7) [inline plugins/servicenow/plugin.spec.yaml:704]:
```suggestion
        description: The name of source DNS where the vulnerability was found
```

File: plugins/servicenow/plugin.spec.yaml [hunk 4]
```diff
@@ -670,12 +681,42 @@ actions:
...
       first_found:
         title: First Found
         description: The time that represents the vulnerability was first found, in ISO format
         type: date
-        required: true
+        required: false
         example: 2023-04-28T15:48:07
+      last_found:
+        title: Last Found
+        description: The time that represents the vulnerability was last found, in ISO format
```
Comment (ablakley-r7) [inline plugins/servicenow/plugin.spec.yaml:716]:
```suggestion
        description: The time that represents when the vulnerability was last found, in ISO format
```

File: plugins/servicenow/plugin.spec.yaml [hunk 5]
```diff
@@ -771,6 +800,12 @@ actions:
           - ""
         required: false
         example: SAFE
+      vulnerability:
+        title: Vulnerability
+        description: The reference of vulnerability found
```
Comment (ablakley-r7) [inline plugins/servicenow/plugin.spec.yaml:805]:
```suggestion
        description: The reference of the found vulnerability
```

File: plugins/servicenow/plugin.spec.yaml [hunk 6]
```diff
@@ -780,12 +815,36 @@ actions:
...
         example: Open
+      source:
+        title: Source
+        description: The vulnerability source
+        type: string
+        required: false
+        example: ExampleSource
+      ip_address:
+        title: IP Address
+        description: The IP address of source the vulnerability was found
```
Comment (ablakley-r7) [inline plugins/servicenow/plugin.spec.yaml:826]:
```suggestion
        description: The IP address of the source where the vulnerability was found
```

File: plugins/servicenow/plugin.spec.yaml [hunk 7]
```diff
@@ -780,12 +815,36 @@ actions:
...
+        type: string
+        required: false
+        example: 192.168.0.1
+      dns:
+        title: DNS Name
+        description: The name of source DNS the vulnerability was found
```
Comment (ablakley-r7) [inline plugins/servicenow/plugin.spec.yaml:832]:
```suggestion
        description: The name of the source DNS where the vulnerability was found
```

File: plugins/servicenow/plugin.spec.yaml [hunk 8]
```diff
@@ -780,12 +815,36 @@ actions:
...
+        example: dns.example.com
       first_found:
         title: First Found
         description: The time that represents the vulnerability was first found, in ISO format
         type: date
         required: false
         example: 2023-04-28T15:48:07
+      last_found:
+        title: Last Found
+        description: The time that represents the vulnerability was last found, in ISO format
```
Comment (ablakley-r7) [inline plugins/servicenow/plugin.spec.yaml:844]:
```suggestion
        description: The time that represents when the vulnerability was last found, in ISO format
```

File: plugins/servicenow/plugin.spec.yaml [hunk 9]
```diff
@@ -919,9 +966,9 @@ triggers:
         default: 5
         example: 5
     output:
-      changed_fields:
+      updated_vulnerabilities:
         title: Changed Fields
-        description: JSON object representing changed fields (map of field name to previous and current values)
-        type: object
+        description: List of JSON object containing system ID of updated vulnerability and representing changed fields (map of field name to previous and current values)
```
Comment (ablakley-r7) [inline plugins/servicenow/plugin.spec.yaml:971]:
```suggestion
        description: List of JSON objects containing the system ID of the updated vulnerability and representation of the changed fields (map of field name from previous to current values)
```

**Review by ablakley-r7 — CHANGES_REQUESTED** [review]
Changes look great! I hope you don't mind, I have suggested rephrasing some of the output/input descriptions.

**Review by cmcnally-r7 — COMMENTED** [review]
Should the plugin version also be increased?

**Comment by igorski-r7** [comment]
> Should the plugin version also be increased?

It was bumped in the previous PR. This is the continuation 😄

**Review by cmcnally-r7 — APPROVED** [review]

**Pull request merged**